Reword introduction and explanation of codegen (#4635)
digital-asset/daml: docs/source/getting-started/index.rst

.. _new-quickstart:

Getting Started with DAML
#########################

**Disclaimer:** This guide is being actively developed.
Expect major changes to the tutorial text and minor changes to the template application.

The goal of this tutorial is to get you up and running with full-stack DAML development.
We do this through the example of a simple social networking application,
showing you three things:

    1. How to build and run the application
    2. The design of its different components
    3. How to write a new feature for the app.

> We do not aim to be comprehensive in all DAML concepts and tools (covered in Writing DAML) or in all deployment options (see Deploying).

Reviewer: `Writing DAML` and `Deploying` should be links.

The goal is that by the end of this tutorial, you'll have a good idea of the following:

    1. What DAML contracts and ledgers are
    2. How a user interface (UI) interacts with a DAML ledger
    3. How DAML helps you build a real-life application fast.

With that, let's get started!

> .. TODO: link to sections of this guide

Reviewer: oh I see, well I’m fine with keeping it for another PR but might be easier to just include it here to avoid having to go through CI twice for such a small change.

    git clone https://github.com/digital-asset/create-daml-app.git
    cd create-daml-app

> Next we need to compile the DAML code to an archive ``.dar`` file::

Reviewer: ```suggestion
Next we need to compile the DAML code to a DAR file::
```

> Next we need to compile the DAML code to an archive ``.dar`` file::

Reviewer 2: ```suggestion
Next we need to compile the DAML code to a DAML archive ``.dar`` file::
```

    daml build

> In order to connect the UI code to this DAML, we need to run a code generation step (which we'll explain later)::

Reviewer 2: I find the "I'll explain this later" stance problematic. Let's try to remove in a future iteration again.

Author: Removed
